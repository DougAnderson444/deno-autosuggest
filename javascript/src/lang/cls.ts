
export const ANY         = 0
export const CONTROL     = 1
export const WHITESPACE  = 2
export const PUNCTUATION = 3
export const NOTALPHA    = 4
export const NOTALPHANUM = 5
export const CONSONANT   = 6
export const VOWEL       = 7


export type CharClass =
    | typeof ANY
    | typeof CONTROL
    | typeof WHITESPACE
    | typeof PUNCTUATION
    | typeof NOTALPHA
    | typeof NOTALPHANUM
    | typeof CONSONANT
    | typeof VOWEL

