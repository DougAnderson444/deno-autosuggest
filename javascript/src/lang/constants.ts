import {CharClass, CONSONANT, VOWEL} from './cls'

export const CHAR_CLASSES_LATIN: [CharClass, string][] = [
    [CONSONANT, 'b'],
    [CONSONANT, 'c'],
    [CONSONANT, 'd'],
    [CONSONANT, 'f'],
    [CONSONANT, 'g'],
    [CONSONANT, 'h'],
    [CONSONANT, 'j'],
    [CONSONANT, 'k'],
    [CONSONANT, 'l'],
    [CONSONANT, 'm'],
    [CONSONANT, 'n'],
    [CONSONANT, 'p'],
    [CONSONANT, 'q'],
    [CONSONANT, 'r'],
    [CONSONANT, 's'],
    [CONSONANT, 't'],
    [CONSONANT, 'v'],
    [CONSONANT, 'w'],
    [CONSONANT, 'x'],
    [CONSONANT, 'z'],
    [VOWEL, 'a'],
    [VOWEL, 'e'],
    [VOWEL, 'i'],
    [VOWEL, 'o'],
    [VOWEL, 'u'],
    [VOWEL, 'y'],
]
