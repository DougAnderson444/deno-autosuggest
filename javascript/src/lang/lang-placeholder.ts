import {LangAbstract} from './lang'
import {Stemmer} from './stemming/ext/PlaceholderStemmer'

export class Lang extends LangAbstract {
    constructor() {
        super(
            new Stemmer(),
            [],
            [],
            [],
            [],
        )
    }
}
