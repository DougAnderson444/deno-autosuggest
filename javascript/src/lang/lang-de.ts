import {LangAbstract} from './lang'
import {Stemmer} from './stemming/ext/GermanStemmer'
import {CHAR_CLASSES_LATIN} from './constants'
import {CharClass, CONSONANT} from './cls'
import {
    PartOfSpeech,
    ARTICLE,
    PREPOSITION,
    CONJUNCTION,
    PARTICLE,
} from './pos'


const FUNCTION_WORDS: [PartOfSpeech, string][] = [
    [ARTICLE, "das"],
    [ARTICLE, "dem"],
    [ARTICLE, "den"],
    [ARTICLE, "der"],
    [ARTICLE, "des"],
    [ARTICLE, "die"],
    [ARTICLE, "ein"],
    [ARTICLE, "eine"],
    [ARTICLE, "einem"],
    [ARTICLE, "einen"],
    [ARTICLE, "einer"],
    [ARTICLE, "eines"],

    [PREPOSITION, "an"],
    [PREPOSITION, "auf"],
    [PREPOSITION, "aus"],
    [PREPOSITION, "bei"],
    [PREPOSITION, "bis"],
    [PREPOSITION, "durch"],
    [PREPOSITION, "entlang"],
    [PREPOSITION, "für"],
    [PREPOSITION, "gegen"],
    [PREPOSITION, "hinter"],
    [PREPOSITION, "in"],
    [PREPOSITION, "mit"],
    [PREPOSITION, "nach"],
    [PREPOSITION, "neben"],
    [PREPOSITION, "ohne"],
    [PREPOSITION, "seit"],
    [PREPOSITION, "um"],
    [PREPOSITION, "von"],
    [PREPOSITION, "zu"],

    [CONJUNCTION, "aber"],
    [CONJUNCTION, "als"],
    [CONJUNCTION, "als"],
    [CONJUNCTION, "anstatt"],
    [CONJUNCTION, "auch"],
    [CONJUNCTION, "bevor"],
    [CONJUNCTION, "bis"],
    [CONJUNCTION, "but"],
    [CONJUNCTION, "damit"],
    [CONJUNCTION, "dass"],
    [CONJUNCTION, "denn"],
    [CONJUNCTION, "entweder"],
    [CONJUNCTION, "nachdem"],
    [CONJUNCTION, "noch"],
    [CONJUNCTION, "ob"],
    [CONJUNCTION, "obwohl"],
    [CONJUNCTION, "oder"],
    [CONJUNCTION, "oder"],
    [CONJUNCTION, "seitdem"],
    [CONJUNCTION, "sobald"],
    [CONJUNCTION, "sofern"],
    [CONJUNCTION, "sondern"],
    [CONJUNCTION, "soweit"],
    [CONJUNCTION, "sowie"],
    [CONJUNCTION, "sowohl"],
    [CONJUNCTION, "sowohl"],
    [CONJUNCTION, "the"],
    [CONJUNCTION, "und"],
    [CONJUNCTION, "während"],
    [CONJUNCTION, "weder"],
    [CONJUNCTION, "weil"],
    [CONJUNCTION, "wenn"],
    [CONJUNCTION, "wie"],
    [CONJUNCTION, "wie"],
    [CONJUNCTION, "wo"],
    [CONJUNCTION, "zu"],

    [PARTICLE, "schon"],
    [PARTICLE, "ja"],
    [PARTICLE, "halt"],
    [PARTICLE, "wohl"],
    [PARTICLE, "doch"],
    [PARTICLE, "mal"],
    [PARTICLE, "aber"],
    [PARTICLE, "auch"],
    [PARTICLE, "bloß"],
    [PARTICLE, "denn"],
    [PARTICLE, "eben"],
    [PARTICLE, "etwas"],
    [PARTICLE, "nur"],
    [PARTICLE, "ruhig"],
    [PARTICLE, "shon"],
    [PARTICLE, "zwar"],
    [PARTICLE, "soweiso"],
]


const CHAR_CLASSES: [CharClass, string][] = [
    [CONSONANT, 'ß'],
]


const UTF_COMPOSE_MAP: [string, string][] = [
    ["Ä", "Ä"],
    ["Ö", "Ö"],
    ["Ü", "Ü"],
    ["ä", "ä"],
    ["ö", "ö"],
    ["ü", "ü"],
]


const UTF_REDUCE_MAP: [string, string][] = [
    ["ẞ", "SS"], // eszett
    ["ß", "ss"],
    ["Ä", "A"], // umlauts
    ["Ö", "O"],
    ["Ü", "U"],
    ["ä", "a"],
    ["ö", "o"],
    ["ü", "u"],
]


export class Lang extends LangAbstract {
    constructor() {
        super(
            new Stemmer(),
            [...CHAR_CLASSES_LATIN, ...CHAR_CLASSES],
            FUNCTION_WORDS,
            UTF_COMPOSE_MAP,
            UTF_REDUCE_MAP,
        )
    }
}
