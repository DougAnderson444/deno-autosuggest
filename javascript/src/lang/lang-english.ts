import {Lang} from './lang'
import {EnglishStemmer} from './stemming/ext/EnglishStemmer'
import {CHAR_CLASSES_LATIN} from './constants'
import {CharClass} from './cls'
import {
    PartOfSpeech,
    ARTICLE,
    PREPOSITION,
    CONJUNCTION,
    PARTICLE,
} from './pos'


const FUNCTION_WORDS: [PartOfSpeech, string][] = [
    [ARTICLE, "a"],
    [ARTICLE, "an"],
    [ARTICLE, "the"],

    [PREPOSITION, "at"],
    [PREPOSITION, "by"],
    [PREPOSITION, "for"],
    [PREPOSITION, "from"],
    [PREPOSITION, "in"],
    [PREPOSITION, "of"],
    [PREPOSITION, "on"],
    [PREPOSITION, "to"],
    [PREPOSITION, "since"],
    [PREPOSITION, "before"],
    [PREPOSITION, "till"],
    [PREPOSITION, "untill"],
    [PREPOSITION, "beside"],
    [PREPOSITION, "under"],
    [PREPOSITION, "below"],
    [PREPOSITION, "over"],
    [PREPOSITION, "above"],
    [PREPOSITION, "across"],
    [PREPOSITION, "through"],
    [PREPOSITION, "into"],
    [PREPOSITION, "towards"],
    [PREPOSITION, "onto"],
    [PREPOSITION, "off"],
    [PREPOSITION, "out"],
    [PREPOSITION, "about"],

    [CONJUNCTION, "after"],
    [CONJUNCTION, "although"],
    [CONJUNCTION, "as"],
    [CONJUNCTION, "because"],
    [CONJUNCTION, "before"],
    [CONJUNCTION, "but"],
    [CONJUNCTION, "either"],
    [CONJUNCTION, "for"],
    [CONJUNCTION, "how"],
    [CONJUNCTION, "if"],
    [CONJUNCTION, "lest"],
    [CONJUNCTION, "nor"],
    [CONJUNCTION, "once"],
    [CONJUNCTION, "once"],
    [CONJUNCTION, "or"],
    [CONJUNCTION, "since"],
    [CONJUNCTION, "since"],
    [CONJUNCTION, "since"],
    [CONJUNCTION, "so"],
    [CONJUNCTION, "than"],
    [CONJUNCTION, "that"],
    [CONJUNCTION, "that"],
    [CONJUNCTION, "though"],
    [CONJUNCTION, "till"],
    [CONJUNCTION, "unless"],
    [CONJUNCTION, "until"],
    [CONJUNCTION, "until"],
    [CONJUNCTION, "until"],
    [CONJUNCTION, "what"],
    [CONJUNCTION, "whatever"],
    [CONJUNCTION, "when"],
    [CONJUNCTION, "when"],
    [CONJUNCTION, "whenever"],
    [CONJUNCTION, "where"],
    [CONJUNCTION, "whereas"],
    [CONJUNCTION, "whereas"],
    [CONJUNCTION, "wherever"],
    [CONJUNCTION, "whether"],
    [CONJUNCTION, "which"],
    [CONJUNCTION, "whichever"],
    [CONJUNCTION, "while"],
    [CONJUNCTION, "while"],
    [CONJUNCTION, "while"],
    [CONJUNCTION, "whilst"],
    [CONJUNCTION, "who"],
    [CONJUNCTION, "whoever"],
    [CONJUNCTION, "whom"],
    [CONJUNCTION, "whomever"],
    [CONJUNCTION, "whose"],
    [CONJUNCTION, "why"],
    [CONJUNCTION, "yet"],
    [CONJUNCTION, "and"],
    // [CONJUNCTION, "as if"],
    // [CONJUNCTION, "as long as"],
    // [CONJUNCTION, "as much as"],
    // [CONJUNCTION, "as soon as"],
    // [CONJUNCTION, "as though"],
    // [CONJUNCTION, "assuming that"],
    // [CONJUNCTION, "by the time"],
    // [CONJUNCTION, "even if"],
    // [CONJUNCTION, "even though"],
    // [CONJUNCTION, "in case that"],
    // [CONJUNCTION, "in case"],
    // [CONJUNCTION, "in order that"],
    // [CONJUNCTION, "in order"],
    // [CONJUNCTION, "now that"],
    // [CONJUNCTION, "only if"],
    // [CONJUNCTION, "provided that"],
    // [CONJUNCTION, "rather than"],
    // [CONJUNCTION, "so that"],

    [PARTICLE, "by"],
    [PARTICLE, "in"],
    [PARTICLE, "not"],
    [PARTICLE, "on"],
    [PARTICLE, "to"],
    [PARTICLE, "oh"],
]


const CHAR_CLASSES: [CharClass, string][] = [
]


const UTF_COMPOSE_MAP: [string, string][] = [
]


const UTF_REDUCE_MAP: [string, string][] = [
]


export const LANG_ENGLISH = new Lang(
    new EnglishStemmer(),
    [...CHAR_CLASSES_LATIN, ...CHAR_CLASSES],
    FUNCTION_WORDS,
    UTF_COMPOSE_MAP,
    UTF_REDUCE_MAP,
)
