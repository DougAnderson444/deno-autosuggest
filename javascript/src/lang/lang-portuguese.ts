import {Lang} from './lang'
import {PortugueseStemmer} from './stemming/ext/PortugueseStemmer'
import {CHAR_CLASSES_LATIN} from './constants'
import {CharClass} from './cls'
import {
    PartOfSpeech,
    ARTICLE,
    PREPOSITION,
    CONJUNCTION,
} from './pos'


const FUNCTION_WORDS: [PartOfSpeech, string][] = [
    [ARTICLE, "o"],
    [ARTICLE, "a"],
    [ARTICLE, "os"],
    [ARTICLE, "as"],
    [ARTICLE, "um"],
    [ARTICLE, "uma"],
    [ARTICLE, "uns"],
    [ARTICLE, "umas"],

    [PREPOSITION, "abaixo"],
    [PREPOSITION, "acima"],
    [PREPOSITION, "além"],
    [PREPOSITION, "antes"],
    [PREPOSITION, "aproximadamente"],
    [PREPOSITION, "aquela"],
    [PREPOSITION, "aquele"],
    [PREPOSITION, "aqueles"],
    [PREPOSITION, "até"],
    [PREPOSITION, "atrás"],
    [PREPOSITION, "com"],
    [PREPOSITION, "como"],
    [PREPOSITION, "conforme"],
    [PREPOSITION, "contra"],
    [PREPOSITION, "de"],
    [PREPOSITION, "depois"],
    [PREPOSITION, "desde"],
    [PREPOSITION, "distante"],
    [PREPOSITION, "durante"],
    [PREPOSITION, "em"],
    [PREPOSITION, "entre"],
    [PREPOSITION, "esta"],
    [PREPOSITION, "estas"],
    [PREPOSITION, "este"],
    [PREPOSITION, "estes"],
    [PREPOSITION, "exceto"],
    [PREPOSITION, "fora"],
    [PREPOSITION, "mais"],
    [PREPOSITION, "mas"],
    [PREPOSITION, "oposto"],
    [PREPOSITION, "para"],
    [PREPOSITION, "perto"],
    [PREPOSITION, "por"],
    [PREPOSITION, "próximo"],
    [PREPOSITION, "que"],
    [PREPOSITION, "sem"],
    [PREPOSITION, "sob"],
    [PREPOSITION, "sobre"],
    [PREPOSITION, "via"],
    // [PREPOSITION, "além de"],
    // [PREPOSITION, "antes de"],
    // [PREPOSITION, "ao lado de"],
    // [PREPOSITION, "apesar de"],
    // [PREPOSITION, "apesar de"],
    // [PREPOSITION, "através de"],
    // [PREPOSITION, "através de"],
    // [PREPOSITION, "bem como"],
    // [PREPOSITION, "dentro de"],
    // [PREPOSITION, "dentro de"],
    // [PREPOSITION, "dentro de"],
    // [PREPOSITION, "devido a"],
    // [PREPOSITION, "diferente de"],
    // [PREPOSITION, "em cima de"],
    // [PREPOSITION, "em direção a"],
    // [PREPOSITION, "em nome de"],
    // [PREPOSITION, "em torno de"],
    // [PREPOSITION, "em vez de"],
    // [PREPOSITION, "fora de"],
    // [PREPOSITION, "fora de"],
    // [PREPOSITION, "longe de"],
    // [PREPOSITION, "na frente de"],
    // [PREPOSITION, "para baixo"],
    // [PREPOSITION, "para cima"],
    // [PREPOSITION, "perto de"],
    // [PREPOSITION, "por causa de"],
    // [PREPOSITION, "próximo a"],
    // [PREPOSITION, "próximo de"],
    // [PREPOSITION, "tanto quanto"],

    [CONJUNCTION, "agora"],
    [CONJUNCTION, "como"],
    [CONJUNCTION, "contudo"],
    [CONJUNCTION, "e"],
    [CONJUNCTION, "enquanto"],
    [CONJUNCTION, "então"],
    [CONJUNCTION, "logo"],
    [CONJUNCTION, "mas"],
    [CONJUNCTION, "nem"],
    [CONJUNCTION, "ou"],
    [CONJUNCTION, "para"],
    [CONJUNCTION, "pois"],
    [CONJUNCTION, "por"],
    [CONJUNCTION, "porém"],
    [CONJUNCTION, "porque"],
    [CONJUNCTION, "portanto"],
    [CONJUNCTION, "quando"],
    [CONJUNCTION, "se"],
    [CONJUNCTION, "todavia"],
    // [CONJUNCTION, "a fim de"],
    // [CONJUNCTION, "ainda assim"],
    // [CONJUNCTION, "ainda que"],
    // [CONJUNCTION, "apesar disso"],
    // [CONJUNCTION, "assim que"],
    // [CONJUNCTION, "como se"],
    // [CONJUNCTION, "enquanto que"],
    // [CONJUNCTION, "não só ... como também"],
    // [CONJUNCTION, "no entanto"],
    // [CONJUNCTION, "ora ... ora"],
    // [CONJUNCTION, "ou ... ou"],
    // [CONJUNCTION, "por conseguinte"],
    // [CONJUNCTION, "por isso"],
    // [CONJUNCTION, "quer ... quer"],
    // [CONJUNCTION, "sempre que"],
    // [CONJUNCTION, "tanto ... como"],
    // [CONJUNCTION, "visto que"],
]


const CHAR_CLASSES: [CharClass, string][] = [
]


const UTF_COMPOSE_MAP: [string, string][] = [
    ["Ç", "Ç"], // cedilla
    ["ç", "ç"],
    ["Á", "Á"], // acute accent
    ["É", "É"],
    ["Í", "Í"],
    ["Ó", "Ó"],
    ["Ú", "Ú"],
    ["á", "á"],
    ["é", "é"],
    ["í", "í"],
    ["ó", "ó"],
    ["ú", "ú"],
    ["Â", "Â"], // circumflex accent
    ["Ê", "Ê"],
    ["Ô", "Ô"],
    ["â", "â"],
    ["ê", "ê"],
    ["ô", "ô"],
    ["Ã", "Ã"], // tilde
    ["Õ", "Õ"],
    ["ã", "ã"],
    ["õ", "õ"],
    ["À", "À"], // grave accent
    ["È", "È"],
    ["Ì", "Ì"],
    ["Ò", "Ò"],
    ["Ù", "Ù"],
    ["à", "à"],
    ["è", "è"],
    ["ì", "ì"],
    ["ò", "ò"],
    ["ù", "ù"],
]


const UTF_REDUCE_MAP: [string, string][] = [
    ["Ç", "C"], // cedilla
    ["ç", "c"],
    ["Á", "A"], // acute accent
    ["É", "E"],
    ["Í", "I"],
    ["Ó", "O"],
    ["Ú", "U"],
    ["á", "a"],
    ["é", "e"],
    ["í", "i"],
    ["ó", "o"],
    ["ú", "u"],
    ["Â", "A"], // circumflex accent
    ["Ê", "E"],
    ["Ô", "O"],
    ["â", "a"],
    ["ê", "e"],
    ["ô", "o"],
    ["Ã", "A"], // tilde
    ["Õ", "O"],
    ["ã", "a"],
    ["õ", "o"],
    ["À", "A"], // grave accent
    ["È", "E"],
    ["Ì", "I"],
    ["Ò", "O"],
    ["Ù", "U"],
    ["à", "a"],
    ["è", "e"],
    ["ì", "i"],
    ["ò", "o"],
    ["ù", "u"],
]


export const LANG_PORTUGUESE = new Lang(
    new PortugueseStemmer(),
    [...CHAR_CLASSES_LATIN, ...CHAR_CLASSES],
    FUNCTION_WORDS,
    UTF_COMPOSE_MAP,
    UTF_REDUCE_MAP,
)
