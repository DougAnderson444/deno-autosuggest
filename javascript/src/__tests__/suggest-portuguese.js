import {LucidSuggest} from "../index"
import {LANG_PORTUGUESE} from "../lang/lang-portuguese"


describe("Portuguese language", () => {
    const records = [
        {id: 10, title: "Pilhas Alcalinas AA — Embalagem com 12"},
        {id: 20, title: "Cabo Lightning para USB-A"},
        {id: 30, title: "Escova de dentes elétrica"},
        {id: 40, title: "Sacos de armazenamento a vácuo"},
    ]

    const suggest = new LucidSuggest(LANG_PORTUGUESE)
    suggest.addRecords(records)

    test("Empty input", async () => {
        const hits = await suggest.search("")
        expect(hits).toMatchSnapshot()
    })

    test("Equality", async () => {
        const hits = await suggest.search("escova de dentes eletrica")
        expect(hits).toMatchSnapshot()
    })

    test("Stemming", async () => {
        const hits = await suggest.search("alcalino")
        expect(hits).toMatchSnapshot()
    })

    test("Partiles", async () => {
        const hits = await suggest.search("de")
        expect(hits).toMatchSnapshot()
    })

    test("Normalization", async () => {
        const hits = await suggest.search("electric")
        expect(hits).toMatchSnapshot()
    })
})