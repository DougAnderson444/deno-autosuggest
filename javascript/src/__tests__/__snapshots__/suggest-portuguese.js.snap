// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Portuguese language Empty input 1`] = `
Array [
  Hit {
    "chunks": Array [
      Object {
        "highlight": false,
        "text": "Escova de dentes elétrica",
      },
    ],
    "record": Object {
      "id": 30,
      "title": "Escova de dentes elétrica",
    },
  },
  Hit {
    "chunks": Array [
      Object {
        "highlight": false,
        "text": "Cabo Lightning para USB-A",
      },
    ],
    "record": Object {
      "id": 20,
      "title": "Cabo Lightning para USB-A",
    },
  },
  Hit {
    "chunks": Array [
      Object {
        "highlight": false,
        "text": "Sacos de armazenamento a vácuo",
      },
    ],
    "record": Object {
      "id": 40,
      "title": "Sacos de armazenamento a vácuo",
    },
  },
  Hit {
    "chunks": Array [
      Object {
        "highlight": false,
        "text": "Pilhas Alcalinas AA — Embalagem com 12",
      },
    ],
    "record": Object {
      "id": 10,
      "title": "Pilhas Alcalinas AA — Embalagem com 12",
    },
  },
]
`;

exports[`Portuguese language Equality 1`] = `
Array [
  Hit {
    "chunks": Array [
      Object {
        "highlight": true,
        "text": "Escova",
      },
      Object {
        "highlight": false,
        "text": " ",
      },
      Object {
        "highlight": true,
        "text": "de",
      },
      Object {
        "highlight": false,
        "text": " ",
      },
      Object {
        "highlight": true,
        "text": "dentes",
      },
      Object {
        "highlight": false,
        "text": " ",
      },
      Object {
        "highlight": true,
        "text": "elétrica",
      },
    ],
    "record": Object {
      "id": 30,
      "title": "Escova de dentes elétrica",
    },
  },
  Hit {
    "chunks": Array [
      Object {
        "highlight": false,
        "text": "Sacos ",
      },
      Object {
        "highlight": true,
        "text": "de",
      },
      Object {
        "highlight": false,
        "text": " armazenamento a vácuo",
      },
    ],
    "record": Object {
      "id": 40,
      "title": "Sacos de armazenamento a vácuo",
    },
  },
]
`;

exports[`Portuguese language Normalization 1`] = `
Array [
  Hit {
    "chunks": Array [
      Object {
        "highlight": false,
        "text": "Escova de dentes ",
      },
      Object {
        "highlight": true,
        "text": "elétric",
      },
      Object {
        "highlight": false,
        "text": "a",
      },
    ],
    "record": Object {
      "id": 30,
      "title": "Escova de dentes elétrica",
    },
  },
]
`;

exports[`Portuguese language Partiles 1`] = `
Array [
  Hit {
    "chunks": Array [
      Object {
        "highlight": false,
        "text": "Escova de ",
      },
      Object {
        "highlight": true,
        "text": "de",
      },
      Object {
        "highlight": false,
        "text": "ntes elétrica",
      },
    ],
    "record": Object {
      "id": 30,
      "title": "Escova de dentes elétrica",
    },
  },
  Hit {
    "chunks": Array [
      Object {
        "highlight": false,
        "text": "Sacos ",
      },
      Object {
        "highlight": true,
        "text": "de",
      },
      Object {
        "highlight": false,
        "text": " armazenamento a vácuo",
      },
    ],
    "record": Object {
      "id": 40,
      "title": "Sacos de armazenamento a vácuo",
    },
  },
]
`;

exports[`Portuguese language Stemming 1`] = `
Array [
  Hit {
    "chunks": Array [
      Object {
        "highlight": false,
        "text": "Pilhas ",
      },
      Object {
        "highlight": true,
        "text": "Alcalin",
      },
      Object {
        "highlight": false,
        "text": "as AA — Embalagem com 12",
      },
    ],
    "record": Object {
      "id": 10,
      "title": "Pilhas Alcalinas AA — Embalagem com 12",
    },
  },
]
`;
